torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1433,6 +1433,37 @@ class ShapePropagator {
           dtype, device, dim, /*requires_grad=*/c10::nullopt)};
     };
 
+    static const auto factory_like_with_ndim = [](Node* node,
+                                                  int dim) -> type_vec_t {
+      at::optional<IValue> maybe_layout_option = node->get(attr::layout);
```

Comment:
does layout affect the type ? why are we bailing if it's not a constant but not using it to create the type